internal/mux/buffer.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2019 The Oto Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mux
+
+import (
+	"bytes"
+	"sync"
+)
+
+// NewConcurrentBuffer returns a ready-to-use ConcurrentBuffer.
+func NewConcurrentBuffer() ConcurrentBuffer {
+	return ConcurrentBuffer{
+		buf: &bytes.Buffer{},
+		m:   &sync.Mutex{},
+	}
+}
+
+// ConcurrentBuffer provides a bytes.Buffer that is safe for concurrent use.
+type ConcurrentBuffer struct {
+	buf *bytes.Buffer
+	m   *sync.Mutex
```

Comment:
Use `bytes.Buffer` and `sync.Mutex` instead of pointers.

Reply:
I'll make this change and then update the methods to be pointer receivers as well.